authorize/grpc_test.go

```diff
@@ -142,13 +158,144 @@ func Test_getEvaluatorRequestWithPortInHostHeader(t *testing.T) {
 				"Accept":            "text/html",
 				"X-Forwarded-Proto": "https",
 			},
-			certPEM,
+			evaluator.ClientCertificateInfo{},
 			"",
 		),
 	}
 	assert.Equal(t, expect, actual)
 }
 
+func Test_getClientCertificateInfo(t *testing.T) {
+	const leafPEM = `-----BEGIN CERTIFICATE-----
```

Comment:
optional: I think we can add `t.Parallel()`.

Reply:
I wanted to also verify the log output from the `getClientCertificateInfo()` method, which is why I added a new `testutil.SetLogger()` method. If I'm understanding correctly, I think we need to make sure multiple parallel tests don't each try to set their own logger (as the logger is shared global state). Let me know if I'm misunderstanding how parallel tests work.